src/charm.py

```diff
@@ -513,25 +397,7 @@ def _log_charm_status(self) -> None:
     @catch_charm_errors
     def _on_stop(self, _: StopEvent) -> None:
         """Handle the stopping of the charm."""
-        self._event_timer.disable_event_timer("reconcile-runners")
-        state = self._setup_state()
-        runner_scaler = create_runner_scaler(state, self.app.name, self.unit.name)
-        runner_scaler.flush(FlushMode.FLUSH_BUSY)
-
-    def _reconcile_openstack_runners(self, runner_scaler: RunnerScaler) -> None:
-        """Reconcile the current runners state and intended runner state for OpenStack mode.
-
-        Args:
-            runner_scaler: Scaler used to scale the amount of runners.
-        """
-        self.unit.status = MaintenanceStatus("Reconciling runners")
-        try:
-            runner_scaler.reconcile()
-        except ReconcileError:
-            logger.exception(FAILED_TO_RECONCILE_RUNNERS_MSG)
-            self.unit.status = ActiveStatus(ACTIVE_STATUS_RECONCILIATION_FAILED_MSG)
-        else:
-            self.unit.status = ActiveStatus()
+        self._manager_client.flush_runner(busy=True)
```

Comment:
Wouldn't we also need to stop the service when we stop it to avoid new resources being created?